src/jti_content.js

```diff
@@ -1,169 +1,244 @@
+        if (ctrlDown && (e.keyCode === backtickKey)) { // if ctrl is pressed
+            var tagStartIndex = []; // store index of <TI>
+            var tagEndIndex = []; // store index of </TI>
+            var tagindex = getAllIndexes($(this).val(), tagStartIndex, tagEndIndex); // find all <TI> and </TI> tags in selected template.
+            tagStartIndex = tagindex.start;
+            tagEndIndex = tagindex.end;
+            if (tagStartIndex.length !== 0 && tagEndIndex.length !== 0) { // works only if the selected template contains any <TI> tag
+                if (selectStart === null && selectEnd === null) { // start from first <TI>
+                    var StartPos = FindNextTI(cursorStart, tagStartIndex, tagEndIndex); // find the starting <TI> tag
+                    $(this)[0].setSelectionRange(StartPos.start, StartPos.end);
+                    selectStart = StartPos.start; // set Start Index
+                    selectEnd = StartPos.end; // set End Index
+                } else { // select next <TI> set
+                    if (tagStartIndex.indexOf(selectStart) === tagStartIndex.length - 1 && tagEndIndex.indexOf(selectEnd) === tagEndIndex.length - 1) { // currently selecting the last set of <TI>,back to first set
+                        $(this)[0].setSelectionRange(tagStartIndex[0], tagEndIndex[0]);
+                        selectStart = tagStartIndex[0];
+                        selectEnd = tagEndIndex[0];
+                    } else {
+                        if (tagStartIndex.indexOf(selectStart) === -1 && tagEndIndex.indexOf(selectEnd) === -1) { // highlighted <TI> tag is modified by user. Now we need search for the next <TI>.
+                            StartPos = FindNextTI(cursorStart, tagStartIndex, tagEndIndex); // find the starting <TI> tag
+                            $(this)[0].setSelectionRange(StartPos.start, StartPos.end);
+                            selectStart = StartPos.start; // set Start Index
+                            selectEnd = StartPos.end; // set End Index
+                        } else {
+                            $(this)[0].setSelectionRange(tagStartIndex[tagStartIndex.indexOf(selectStart) + 1], tagEndIndex[tagEndIndex.indexOf(selectEnd) + 1]); // find next set of <TI>
+                            selectStart = tagStartIndex[tagStartIndex.indexOf(selectStart) + 1];
+                            selectEnd = tagEndIndex[tagEndIndex.indexOf(selectEnd) + 1];
+                        }
+                    }
+                }
+            }
+        }
+    });
+});
+
+// Helper method. Find next <TI> based on cursor position
+function FindNextTI (CursorPos, tagStart, tagEnd) {
+    for (var i = 0; i < tagStart.length; i++) {
+        if (tagStart[i] >= CursorPos) {
+            return { start: tagStart[i], end: tagEnd[i] };
+        }
+    }
+}
+
+// Helper method. Find index(start and end) of all occurrences of a given substring in a string
+function getAllIndexes (str, arr1, arr2) {

```

Comment:
You should use more descriptive variables
arr1 -> startIndexes
arr2 -> endIndexes